evalml/tests/component_tests/decomposer_tests/test_decomposer.py

```diff
@@ -476,6 +483,60 @@ def test_decomposer_determine_periodicity(
         assert 0.95 * period <= ac <= 1.05 * period
 
 
+@pytest.mark.parametrize(
+    "decomposer_child_class",
+    decomposer_list,
+)
+@pytest.mark.parametrize(
+    "nullable_ltype",
+    ["IntegerNullable", "AgeNullable"],
+)
+@pytest.mark.parametrize(
+    "handle_incompatibility",
+    [
+        True,
+        pytest.param(
+            False,
+            marks=pytest.mark.xfail(strict=True, raises=AssertionError),
+        ),
+    ],
+)
+def test_decomposer_determine_periodicity_nullable_type_incompatibility(
+    decomposer_child_class,
+    handle_incompatibility,
+    nullable_ltype,
+    generate_seasonal_data,
+):
+    """Testing that the nullable type incompatibility that caused us to add handling for the Decomposer
+    is still present in pandas. If this test is causing the test suite to fail
+    because the code below no longer raises the expected AssertionError, we should confirm that the nullable
+    types now work for our use case and remove the nullable type handling logic from Decomposer.determine_periodicity.
+    """
+    trend_degree = 2
+    period = 7
+    X, y = generate_seasonal_data(real_or_synthetic="synthetic")(
+        period,
+        trend_degree=trend_degree,
+    )
+
+    # Multiply by 1000 so we can convert to Integer, truncating the rest of the value
+    # while still mainlining trend, period, etc
+    y = y * 1000
+    y = ww.init_series(y.astype(int), logical_type=nullable_ltype)
+
+    if handle_incompatibility:
+        dec = decomposer_child_class(degree=trend_degree, period=period)
+        X, y = dec._handle_nullable_types(X, y)
+
+    numpy_float_data = pd.Series(range(len(y)), dtype="float64")
+    numpy_float_data.iloc[-1] = np.nan
+
+    subtracted_floats = y - numpy_float_data
+
+    dropped_nans = subtracted_floats.dropna()
```

Comment:
can you explain why we're tracking this here?

Reply:
7b71388f0 - let me know if this makes sense to you!